lib/TokenManager.ts

```diff
@@ -424,11 +424,8 @@ export class TokenManager implements TokenManagerInterface {
         return tokens[tokenType];
       })
       .catch(err => {
-        // If renew fails, remove token and emit error
-        if (isRefreshTokenError(err) || err.name === 'OAuthError' || err.name === 'AuthSdkError') {
-          // remove token from storage
-          this.remove(key);
-        }
+        // If renew fails, remove token from storage and emit error
+        this.remove(key);
```

Comment:
In case of renew failure, we should update `authState.isAuthenticated` property to false.
Please let me know your thoughts on this?

Reply:
`AuthStateManager` is subscribed to `EVENT_REMOVED` and will update `authState` accordingly

Reply:
Hi @denysoblohin-okta , `AuthStateManager` is subscribed to EVENT_REMOVED but it's not updating the `authState.isAuthenticated` to false. 
Can you please check and let me know ?

Reply:
In the scenario I'm facing this is when the SDK tries to refresh token during offline network.
During the failure of the above case, the `authState.isAuthenticated` remains true. 
If you can check for this case as well, it would be helpful to find the issue.

Reply:
Hi @singh-param-veer 
Tried to reproduce your issue with [the test app](https://github.com/okta/okta-auth-js/tree/master/test/apps/app):
- clone repo, run `yarn build && yarn start`
- open `localhost:8080`, config with correct issuer and client id, sign in
- click `Subscribe to AuthState` and `Subscribe to TokenManager events`
- turn off Internet
- run in console `window._testApp.oktaAuth.tokenManager.renew('accessToken')`
Result:
<img width="1536" alt="Screenshot 2022-02-21 at 11 37 42" src="https://user-images.githubusercontent.com/72614880/154929203-0e473d05-5587-4698-bad1-657fb490281f.png">

Do you have another result? Can you share please?

Reply:
Would it be possible if we can come up on call anytime today for discussing the scenario?

Reply:
Could you reach out to [support team](https://support.okta.com/help/s/?language=en_US) for a call?